### ISSUE TITLE
Through RDP Remote Desktop

### ISSUE THREAD

**Issue by mxmissile** [MarcStan/monogame-framework-wpfinterop#22]
Plain monogame Game works fine through RDP. But WpfGame just shows up white. Works great local. Any ideas?

**Comment by MarcStan** [comment]
I'm not sure what exactly is causing it but to get Monogame into WPF the rendering needs to share resources between DirectX 9 and 11. I wonder if that's too much for the remote machine or RDP has issues detecting the updates from multiple DirectX versions.

Does the remote a machine have a dedicated GPU or is it a regular cloud VM?

I also wonder if other remote solutions suffer the same problems (Skype/Teams/..). I have personally never tried running it over a remote protocol..

**Comment by mxmissile** [comment]
Its at my office, works fine when I'm sitting in front of it. I just dont want to install VS and the whole ceremony on my home computer right now, so I remote in to do dev work. 

My office computer does only have one of those crappy intel gfx chips, so that's probably the reason. 

Its not a big deal, last night I moved all the logic and assets into a library, and now have 2 front ends, one regular monogame and one monogame WPF. This allows me to work on non UI stuff at home in the non WPF version.

Thanks for this library!

**Comment by georg-eckert-zeiss** [comment]
@MarcStan You should be able to fix this by setting `enableSoftwareFallback` in `D3D11Image`:

```csharp
SetBackBuffer( D3DResourceType.IDirect3DSurface9, surface.NativePointer, true );
```

**Comment by rmcfar1999** [comment]
Adding this for posterity: 

While the enableSoftwareFallback is enabled by default in D3D11Image.cs:SetBackBuffer, I also found it necessary to temporarily removing the call to "StopRendering() in the D3D11Host.cs:OnIsFrontBufferAvailableChanged event. 

It appears that when running in RDP the event is called with no FrontBufferAvailable and falls into StopRendering().  I have simply commented out the `else { StopRendering(); }`.  

I am still tracing through the exact issue but have noticed no ill effects as of yet.  

Thanks to this post for setting me on the right path: https://stackoverflow.com/questions/56849171/direct2d-with-wpf-over-rdp
